scripts/mech_events.py

```diff
@@ -0,0 +1,257 @@
+from web3.types import BlockParams
+
+
+SCRIPT_PATH = Path(__file__).resolve().parent
+AGENT_MECH_JSON_PATH = Path(SCRIPT_PATH, "..", "contracts", "AgentMech.json")
+MECH_EVENTS_JSON_PATH = Path(SCRIPT_PATH, "..", ".trader_runner", "mech_events.json")
+HTTP = "http://"
+HTTPS = HTTP[:4] + "s" + HTTP[4:]
+CID_PREFIX = "f01701220"
+IPFS_ADDRESS = f"{HTTPS}gateway.autonolas.tech/ipfs/"
+LATEST_BLOCK_NAME: BlockParams = "latest"
+BLOCK_DATA_NUMBER = "number"
+BLOCKS_CHUNK_SIZE = 5000
+EXCLUDED_BLOCKS_THRESHOLD = 2 * BLOCKS_CHUNK_SIZE
+NUM_EXCLUDED_BLOCKS = 10
+
+# Pair of (Mech contract address, Mech contract deployed on block number).
+MECH_CONTRACT_ADDRESSES = [
+    # Old Mech contract
+    (
+        to_checksum_address("0xff82123dfb52ab75c417195c5fdb87630145ae81"),
+        27939217,
+    ),
+    # New Mech contract
+    (
+        to_checksum_address("0x77af31de935740567cf4ff1986d04b2c964a786a"),
+        30663133,
+    ),
+]
+
+
+@dataclass
+class MechBaseEvent:
+    """Base class for mech's on-chain event representation."""
+
+    event_id: str
+    data: str
+    sender: str
+    transaction_hash: str
+    block_number: int
+    ipfs_link: str
+    ipfs_contents: Dict[str, Any]
+
+    def __init__(
+        self,
+        event_id: str,
+        data: str,
+        sender: str,
+        transaction_hash: str,
+        block_number: int,
+    ):  # pylint: disable=too-many-arguments
+        """Initializes the MechBaseEvent"""
+        self.event_id = event_id
+        self.data = data
+        self.sender = sender
+        self.transaction_hash = transaction_hash
+        self.block_number = block_number
+        self.ipfs_link = ""
+        self.ipfs_contents = {}
+        self._populate_ipfs_contents(data)
+
+    def _populate_ipfs_contents(self, data: str) -> None:
+        url = f"{IPFS_ADDRESS}{CID_PREFIX}{data}"
+        for _url in [f"{url}/metadata.json", url]:
+            try:
+                response = requests.get(_url)
+                response.raise_for_status()
+                self.ipfs_contents = response.json()
+                self.ipfs_link = _url
+            except Exception:  # pylint: disable=broad-except
+                continue
+
+
+@dataclass
+class MechRequest(MechBaseEvent):
+    """A mech's on-chain response representation."""
+
+    request_id: str
+    fee: int
+    event_name: str = "Request"
+
+    def __init__(self, event: AttributeDict):
+        """Initializes the MechRequest"""
+
+        if self.event_name != event["event"]:
+            raise ValueError("Invalid event to initialize MechRequest")
+
+        args = event["args"]
+
+        super().__init__(
+            event_id=args["requestId"],
+            data=args["data"].hex(),
+            sender=args["sender"],
+            transaction_hash=event["transactionHash"].hex(),
+            block_number=event["blockNumber"],
+        )
+
+        self.request_id = self.event_id
+        self.fee = 10000000000000000
```

Comment:
This is the default value of the fees right now. Should be updated in a futue PR as the fee sent to the Mech contract address on the transaction associated to the event (we have the tx_hash).

Reply:
Add a comment in code then please

Reply:
Added.